server/src/main/java/org/opensearch/ingest/IngestPipelineValidator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ingest;
+
+import java.util.List;
+
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Setting;
+
+/**
+ * This class contains methods to validate the ingest pipeline.
+ */
+public class IngestPipelineValidator {
```

Comment:
Any reason for creating this class? Since `IngestService` has the different validation for pipelines as well, let's keep all validation at single place in `IngestService`.

Reply:
Refactored the methods.

Reply:
Updated this.